# Notebook for Inferencing Tuned Model
```
Muhamad Arif Rachman Hatab - Hacktiv8 FTDS-017-RMT
```

# 1 - Import Libraries

In [10]:
# Import Needed Libraries
import pandas as pd
import numpy as np
import json
import pickle

# 2 - Load Files

In [11]:
# Load Files
with open('tuned_svr.pkl', 'rb') as model: # Load Model
    tuned_pipe = pickle.load(model)

with open('selected_feature.txt', 'r') as num: # Load Selected Features
    selected_feature = json.load(num)

with open('data_sample.txt', 'r') as data: # Load Sampled Data
    data_sample = json.load(data)

# Convert Sampled Data to Dataframe
data = pd.read_json(data_sample)
data # Show Dataframe

target        v1       v2      v3     v4        v5        v6       v7  \
131    4.44  658.4375  403.875  696.75  558.5  699.5625  404.5625  725.875   

            v8 sample_type  
131  3961.4375       lab 2

>selected_feature represents list of features with high correlation towards target.

>tuned_pipe is imported tuned version of the model that are trained with the given data, along with the data prepocessing which are scaler and encoder.

>data_sample is a row of data, sampled from the given dataset, in json format, then it needs to be converted to dataframe format before the preprocess and prediction with tuned_pipe.

# 3 - Predict Data Sample

In [12]:
# Predict with Loaded Model
pred = tuned_pipe.predict(data[selected_feature])

In [13]:
# Show Results in Dataframe
data['prediction'] = pred.copy()

# Show Data
data[['target', 'prediction']]

target  prediction
131    4.44    4.526677

The model predict that the target value is at `4.527` which is `.087` apart from the real target value (`4.44`).